Certain backend errors cause requests to not be logged in the audit log

### Description
When debugging the `datetime` issues we noticed that some of the errors being raised were not writing to the Audit log. This also caused an error where the `/graphql` endpoint would return a `400` instead of a `200`

### Steps to reproduce
If applicable, add screenshots to help explain your problem.

1. go to the graphql playground
2. submit a `AddPatientToQueue` mutation with the value `05/14/2021` for `symptomOnset`
3. the API returns a 400 and no items are recorded in the audit log

### Expected behavior
The failed mutation `AddPatientToQueue` is recorded in the audit log and the API returns a `200` with the `CoercingParseValueException` in the response errors

### Additional context
[End-to-end transaction details](https://portal.azure.com/#blade/AppInsightsExtension/DetailsV2Blade/ComponentId/%7B%22SubscriptionId%22%3A%227d1e3999-6577-4cd5-b296-f518e5c8e677%22%2C%22ResourceGroup%22%3A%22prime-simple-report-prod%22%2C%22Name%22%3A%22prime-simple-report-prod-insights%22%2C%22LinkedApplicationType%22%3A0%2C%22ResourceId%22%3A%22%252Fsubscriptions%252F7d1e3999-6577-4cd5-b296-f518e5c8e677%252FresourceGroups%252Fprime-simple-report-prod%252Fproviders%252FMicrosoft.Insights%252Fcomponents%252Fprime-simple-report-prod-insights%22%2C%22ResourceType%22%3A%22microsoft.insights%252Fcomponents%22%2C%22IsAzureFirst%22%3Afalse%7D/DataModel/%7B%22eventId%22%3A%22d87d6d7b-be4d-11eb-8dfa-7f6b3776e47d%22%2C%22timestamp%22%3A%222021-05-26T18%3A11%3A47.424Z%22%2C%22cacheId%22%3A%22fb623b6f-759e-4096-afaf-4059d721b7f3%22%2C%22eventTable%22%3A%22requests%22%7D)
```
Caused by: java.lang.NullPointerException
...
	at gov.cdc.usds.simplereport.logging.AuditLoggingInstrumentation$ExecutionResultContext.onCompleted(AuditLoggingInstrumentation.java:86)
	at gov.cdc.usds.simplereport.logging.AuditLoggingInstrumentation$ExecutionResultContext.onCompleted(AuditLoggingInstrumentation.java:66)
```

--- a/backend/src/main/java/gov/cdc/usds/simplereport/logging/AuditLoggingInstrumentation.java
+++ b/backend/src/main/java/gov/cdc/usds/simplereport/logging/AuditLoggingInstrumentation.java
@@ -79,7 +79,7 @@ public class AuditLoggingInstrumentation extends SimpleInstrumentation {
       LOG.trace("End of execution, audit entry being saved.");
       List<String> errorPaths =
           result.getErrors().stream()
-              .map(e -> e.getPath() == null ? Collections.emptyList() : e.getPath())
+              .map(e -> e.getPath() == null ? new ArrayList<Object>() : e.getPath())
               .flatMap(List::stream)
               .map(Object::toString)
               .collect(Collectors.toList());

--- a/backend/src/main/java/gov/cdc/usds/simplereport/logging/AuditLoggingInstrumentation.java
+++ b/backend/src/main/java/gov/cdc/usds/simplereport/logging/AuditLoggingInstrumentation.java
@@ -8,7 +8,6 @@ import gov.cdc.usds.simplereport.db.model.auxiliary.GraphQlInputs;
 import gov.cdc.usds.simplereport.db.model.auxiliary.HttpRequestDetails;
 import gov.cdc.usds.simplereport.service.AuditService;
 import graphql.ExecutionResult;
-import graphql.GraphQLError;
 import graphql.execution.instrumentation.InstrumentationContext;
 import graphql.execution.instrumentation.InstrumentationState;
 import graphql.execution.instrumentation.SimpleInstrumentation;
@@ -80,7 +79,7 @@ public class AuditLoggingInstrumentation extends SimpleInstrumentation {
       LOG.trace("End of execution, audit entry being saved.");
       List<String> errorPaths =
           result.getErrors().stream()
-              .map(GraphQLError::getPath)
+              .map(e -> e.getPath() == null ? Collections.emptyList() : e.getPath())
               .flatMap(List::stream)
               .map(Object::toString)
               .collect(Collectors.toList());

--- a/backend/src/main/java/gov/cdc/usds/simplereport/logging/AuditLoggingInstrumentation.java
+++ b/backend/src/main/java/gov/cdc/usds/simplereport/logging/AuditLoggingInstrumentation.java
@@ -15,6 +15,7 @@ import graphql.execution.instrumentation.SimpleInstrumentationContext;
 import graphql.execution.instrumentation.parameters.InstrumentationExecutionParameters;
 import graphql.kickstart.servlet.context.GraphQLServletContext;
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 import java.util.stream.Collectors;
 import javax.security.auth.Subject;
@@ -79,7 +80,7 @@ public class AuditLoggingInstrumentation extends SimpleInstrumentation {
       LOG.trace("End of execution, audit entry being saved.");
       List<String> errorPaths =
           result.getErrors().stream()
-              .map(e -> e.getPath() == null ? new ArrayList<Object>() : e.getPath())
+              .map(e -> e.getPath() == null ? Collections.emptyList() : e.getPath())
               .flatMap(List::stream)
               .map(Object::toString)
               .collect(Collectors.toList());

